### ISSUE TITLE
[7.21.0] Antivirus' InfectionFreeFile rule doesn't play well with 'nullable'

### ISSUE THREAD

**Issue by tminich** [aedart/athenaeum#180]
## Description

Mixing `InfictionFreeFile` with the Laravel `nullable` rule leads to an error if the value of the parameter is actually null.
```json
{
    "message": "Unable to open stream for file path null",
    "exception": "Aedart\\Antivirus\\Exceptions\\UnableToOpenFileStream",
    "file": "/var/www/html/vendor/aedart/athenaeum-antivirus/src/Scanners/Concerns/Streams.php",
    "line": 110,
}
```

## Version & Environment

Athenaeum 7.21.0
PHP 8.2.9

## Steps to reproduce

1. Define a rule as `['nullable', 'file', new InfectionFreeFile()]`
2. Send `null` as value

## Expected behaviour

The `null` value is ignored

## Possible solution

`InfectionFreeFile` should check `$value` for null and simply ignore it if it is.

**Comment by aedart** [comment]
Thanks for the report. I will try to solve this during this week.

**Comment by aedart** [comment]
@tminich Im having some trouble to reproduce this. The `nullable` validation rule _SHOULD_ stop any other validation rules from being applied. However, if I submit a string ('null'), then the "_Unable to open stream for file path null_" is indeed thrown, which seems somewhat strange for me... Thought the `file` validation rule should prevent this.

Maybe the problem is something similar to this post: https://laracasts.com/discuss/channels/laravel/problems-using-nullable-and-image-validation-rule-together?page=1&replyId=720417

Would it be possible to either create a small repo that reproduces the issue, so that I can review the code?

**Comment by tminich** [comment]
Ah, the funs of FormData. I thought I had worked around that issue but apparently missed this instance. I'm indeed wondering as well why the `file` rule didn't catch this. The Laravel validations system is, while generally working, imo a bit of a mess... (tangent: Why have an interface for validation rule classes for users but not use it for ANY of the classes that represent rules in your framework?)

Long story short, fixed this on my side, don't think there is anything you could reasonably do about his on yours. 
Well, maybe put the failing file name in quotation marks, that might have clued me in to what was going on. Or even check for `'null'` and throw an Exception with an explanatory message?

**Comment by tminich** [comment]
I did a few tests, the `file` rule fails, but validation continues. Which, in general, makes sense, usually you want to return all issues with a value, not only the first one when validating.

In regards to `InfectionFreeFile` I see two options:
- Act similar to the Laravel `File` rule class, which incorporates the `file` rule (not the same, the class handles some extra stuff). Then you can only proceed with the virus check if the parameter is a valid file.
- Simply fail the rule if the file is not available for checking instead of throwing an exception. (Probably the easier solution)

**Comment by aedart** [comment]
"_[...] Simply fail the rule if the file is not available [...]_", this sounds like an acceptable choice. I will try to make this happen as soon as possible, but it might take a few days (_sorry_).

**Comment by tminich** [comment]
Sounds great. No worries about the time frame though, at least not for me. I have added my general FormData workaround for this end point (as I should have already done a while ago), so the validator actually gets `null` instead of `'null'`, which then works fine.

**Comment by aedart** [comment]
@tminich This defect should not be fixed. Feel free to open another defect/issue, if that isn't the case.
